wrappers/core18.go

```diff
@@ -212,6 +267,86 @@ func writeSnapdServicesOnCore(s *snap.Info, inter interacter) error {
 	return nil
 }
 
+func undoSnapdServicesOnCore(s *snap.Info, sysd systemd.Systemd) error {
```

Comment:
Could you please add a docstring explaining what the function does on a high level?
I'm also wondering if undo* is a good name for these functions (they shouldn't be confused with our undo handlers for tasks).